packages/worker/src/api/controllers/global/tests/configs.spec.ts

```diff
@@ -0,0 +1,146 @@
+import { configs } from "@budibase/backend-core"
+import { UserCtx } from "@budibase/types"
+import * as pro from "@budibase/pro"
+import { find, verifyAIConfig } from "../configs"
+
+jest.mock("@budibase/backend-core", () => ({
+  ...jest.requireActual("@budibase/backend-core"),
+  configs: {
+    getConfig: jest.fn(),
+    save: jest.fn(),
+  },
+}))
+
+describe("Global configs controller", () => {
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it("Should strip secrets when pulling AI config", async () => {
+    configs.getConfig.mockResolvedValue({
+      config: {
+        ai: {
+          apiKey: "abc123APIKey",
+          baseUrl: "https://api.example.com",
+        },
+      },
+    })
+    const ctx = {
+      params: {
+        type: "ai",
+      },
+      throw: jest.fn(),
+    } as UserCtx
+
+    await find(ctx)
+
+    expect(ctx.body).toEqual({
+      config: {
+        ai: {
+          apiKey: "--secret-value--",
+          baseUrl: "https://api.example.com",
+        },
+      },
+    })
+  })
+
+  it("Should return the default BB AI config when the feature is turned on", async () => {
+    pro.features.isBudibaseAIEnabled = jest.fn(() => true)
```

Comment:
Does this assignment need to be reset when this test is finished?

Reply:
Updated to use spies, which will be cleaned up with `resetAllMocks`